src/pyk/ktool/kompile.py

```diff
@@ -114,6 +119,8 @@ def __call__(
             check_dir_path(abs_or_rel_to(include_dir, cwd or Path()))
 
         command = list(command) if command is not None else ['kompile']
+        if shutil.which(command[0]) is None:
+            raise KompileNotFoundException(command[0])
```

Comment:
I'd just go with a `RuntimeError` with a message:
```suggestion
            raise RuntimeError('Kompile command not found: {command[0]}')
```

Reply:
I would like to test in the test that that it actually throws the right exception, without parsing the error message there. So what if I made `KompileNotFoundException` inherit from `RuntimeError`? `KompileNotFoundException.__init__` would take `command[0]` and called `RuntimeError.__init__(f'Kompile command not found: {cmd}')`.

Reply:
Yes, that's better.

> without parsing the error message there

You don't have to, you can just assert equivalence:

```python3
with pytest.raises(RuntimeError) as excinfo:
    ...

assert str(excinfo.value) == 'Kompile command not found: foo'
```